src/services/services.ts

```diff
@@ -6014,7 +6037,7 @@ namespace ts {
                             }
 
                             // Visit the typeReference as well to see if it directly or indirectly use that property
-                            getPropertySymbolsFromBaseTypes(type.symbol, propertyName, result);
+                            getPropertySymbolsFromBaseTypes(type.symbol, propertyName, result, symbol);

```

Comment:
I think you'll just need to pass along a cache for this; also, can you make the `result` parameter come last?